sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -50,7 +50,7 @@ trait FunctionRegistryBase[T] {
   final def registerFunction(
       name: FunctionIdentifier, builder: FunctionBuilder, source: String): Unit = {
     val info = new ExpressionInfo(
-      builder.getClass.getCanonicalName,
+      builder.getClass.getName,
```

Comment:
The change itself is probably fine because it's not used anymore (at least within Spark project) as far as I remember.

But why don't we go more conservative with a check if this is null then use `builder.getClass.getName`.

Reply:
that's a good idea